receiver/statsdreceiver/internal/protocol/metric_translator.go

```diff
@@ -92,6 +93,79 @@ func buildSummaryMetric(desc statsDMetricDescription, summary summaryMetric, sta
 	}
 }
 
+func buildDatadogSummaryMetric(desc statsDMetricDescription, summary summaryMetric, startTime, timeNow time.Time, ilm pmetric.ScopeMetrics) {
+	mCount := ilm.Metrics().AppendEmpty()
+	mCount.SetName(desc.name + ".count")
+	mCountSum := mCount.SetEmptySum()
+	mCountSum.SetIsMonotonic(true)
+	mCountSum.SetAggregationTemporality(pmetric.AggregationTemporalityDelta)
+
+	mAvg := ilm.Metrics().AppendEmpty()
+	mAvg.SetName(desc.name + ".avg")
+	mAvgGauge := mAvg.SetEmptyGauge()
+
+	mMedian := ilm.Metrics().AppendEmpty()
+	mMedian.SetName(desc.name + ".median")
+	mMedianGauge := mMedian.SetEmptyGauge()
+
+	mMax := ilm.Metrics().AppendEmpty()
+	mMax.SetName(desc.name + ".max")
+	mMaxGauge := mMax.SetEmptyGauge()
+
+	mPercentile95 := ilm.Metrics().AppendEmpty()
+	mPercentile95.SetName(desc.name + ".95percentile")
+	mPercentile95Gauge := mPercentile95.SetEmptyGauge()
+
+	numPoints := len(summary.points)
+	rawValues := make([]float64, numPoints)
+	count := float64(0)
+	sum := float64(0)
+	for i := range summary.points {
+		c := summary.weights[i]
+		count += c
+		rawValues[i] = summary.points[i] * c
+		sum += rawValues[i]
+	}
+
+	slices.Sort(rawValues)
+
+	startTimestamp := pcommon.NewTimestampFromTime(startTime)
+	nowTimestamp := pcommon.NewTimestampFromTime(timeNow)
+
+	mCountDp := mCountSum.DataPoints().AppendEmpty()
+	// Note: count is rounded here, see note in counterValue().
+	mCountDp.SetIntValue(int64(count))
```

Comment:
I belive we need to use `math.Round` here as the conversions [spec](https://go.dev/ref/spec#Conversions) under "Conversions between numeric types"  in golang say:

```
When converting a [floating-point number](https://go.dev/ref/spec#Numeric_types) to an integer, the fraction is discarded (truncation towards zero).
```